tests/headers.test.ts

```diff
@@ -0,0 +1,743 @@
+          expect(this.req.headers["x-also-keep"]).toBe("also-keep-this");
+          
+          // Overridden header should have new value
+          expect(this.req.headers["x-override-this"]).toBe("new-value");
+          
+          return [200, { allowed: true }];
+        });
+
+      await fgaClient.check({
+        user: "user:test",
+        relation: "reader",
+        object: "document:test"
+      }, {
+        headers: {
+          "X-Override-This": "new-value"
+        }
+      });
+
+      expect(scope.isDone()).toBe(true);
+    });
+
+    it("should handle case-insensitive header overrides correctly", async () => {
+      const fgaClient = new OpenFgaClient({
+        ...testConfig,
+        baseOptions: {
+          headers: {
+            "X-Test-Header": "default-value"
+          }
+        }
+      });
+
+      const scope = nock(testConfig.apiUrl!)
+        .post(`/stores/${testConfig.storeId}/check`)
+        .reply(function() {
+          // HTTP headers are case-insensitive, so request header should override default
+          const testHeaderValue = this.req.headers["x-test-header"];
+          
+          // Per-request should win
+          expect(testHeaderValue).toBe("request-value");
+          
+          return [200, { allowed: true }];
+        });
+
+      await fgaClient.check({
+        user: "user:test",
+        relation: "reader",
+        object: "document:test"
+      }, {
+        headers: {
+          "x-test-header": "request-value"  // Different case
+        }
+      });
+
+      expect(scope.isDone()).toBe(true);
+    });
+  });
+
+  describe("Content-Type header protection behavior", () => {
+    it("does not honor Content-Type header from baseOptions override", async () => {
+      // The SDK protects Content-Type
+      // User attempts to set Content-Type via baseOptions are ignored
+      
+      const fgaClient = new OpenFgaClient({
+        ...testConfig,
+        baseOptions: {
+          headers: {
+            "Content-Type": "text/plain",           // SDK ignores this
+            "X-Custom-Header": "should-work"        // Custom headers work fine
+          }
+        }
+      });
+
+      const scope = nock(testConfig.apiUrl!)
+        .post(`/stores/${testConfig.storeId}/check`)
+        .reply(function() {
+          const headers = this.req.headers;
+          
+          // SDK enforces Content-Type for JSON APIs
+          expect(headers["content-type"]).toBe("application/json");
+          
+          // Custom headers are preserved
+          expect(headers["x-custom-header"]).toBe("should-work");
+          
+          return [200, { allowed: true }];
+        });
+
+      await fgaClient.check({
+        user: "user:test",
+        relation: "reader",
+        object: "document:test"
+      });
+
+      expect(scope.isDone()).toBe(true);
+    });
+
+    it("allows Content-Type override via per-request headers", async () => {
```

Comment:
I don't know if we actually want this behavior. But this test demonstrates what the actual behavior is; any change to that behavior will be intentional and reflected in the test changes as needed.